lib/vrl/stdlib/src/parse_aws_cloudtrail_logs.rs

```diff
@@ -0,0 +1,1063 @@
+            );
+        }
+        if let Some(edge_device_details) = record.edge_device_details {
+            value.insert("edge_device_details".to_owned(), edge_device_details.into());
+        }
+        if let Some(tls_details) = record.tls_details {
+            value.insert("tls_details".to_owned(), tls_details.into());
+        }
+        if let Some(insight_details) = record.insight_details {
+            value.insert("insight_details".to_owned(), insight_details.into());
+        }
+
+        value.into()
+    }
+}
+
+// Reference: https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-event-reference-user-identity.html.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all(deserialize = "camelCase"))]
+struct AwsCloudTrailLogsRecordUserIdentity {
+    r#type: AwsCloudTrailLogsRecordUserIdentityType,
+    user_name: Option<String>,
+    principal_id: String,
+    arn: String,
+    account_id: String,
+    access_key_id: String,
+    session_context: Option<AwsCloudTrailLogsRecordUserIdentitySessionContext>,
+    invoked_by: Option<String>,
+    identity_provider: Option<String>,
+}
+
+impl From<AwsCloudTrailLogsRecordUserIdentity> for Value {
+    fn from(user_identity: AwsCloudTrailLogsRecordUserIdentity) -> Self {
+        let mut value = BTreeMap::<String, Value>::new();
+
+        value.insert("type".to_owned(), user_identity.r#type.into());
+        if let Some(user_name) = user_identity.user_name {
+            value.insert("user_name".to_owned(), user_name.into());
+        }
+        value.insert("principal_id".to_owned(), user_identity.principal_id.into());
+        value.insert("arn".to_owned(), user_identity.arn.into());
+        value.insert("account_id".to_owned(), user_identity.account_id.into());
+        value.insert(
+            "access_key_id".to_owned(),
+            user_identity.access_key_id.into(),
+        );
+        if let Some(session_context) = user_identity.session_context {
+            value.insert("session_context".to_owned(), session_context.into());
+        }
+        if let Some(invoked_by) = user_identity.invoked_by {
+            value.insert("invoked_by".to_owned(), invoked_by.into());
+        }
+        if let Some(identity_provider) = user_identity.identity_provider {
+            value.insert("identity_provider".to_owned(), identity_provider.into());
+        }
+
+        value.into()
+    }
+}
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
+enum AwsCloudTrailLogsRecordUserIdentityType {
+    Root,
+    #[serde(rename(deserialize = "IAMUser"))]
+    IamUser,
+    AssumedRole,
+    FederatedUser,
+    #[serde(rename(deserialize = "AWSAccount"))]
+    AwsAccount,
+    #[serde(rename(deserialize = "AWSService"))]
+    AwsService,
+}
+
+impl From<AwsCloudTrailLogsRecordUserIdentityType> for Value {
+    fn from(identity_type: AwsCloudTrailLogsRecordUserIdentityType) -> Self {
+        format!("{:?}", identity_type).into()
```

Comment:
I think we should probably implement `Display` rather than using the debug representation.